pwiz_tools/Skyline/Model/Proteome/ProteinMetadataManager.cs

```diff
@@ -200,35 +200,45 @@ private SrmDocument LookupProteinMetadata(SrmDocument docOrig, IProgressMonitor
                                     {
-                                        // That didn't parse well enough to make a search term, or didn't add any new info - just set it as searched so we don't keep trying
-                                        _processedNodes.Add(node.Id.GlobalIndex, proteinMetadata.SetWebSearchCompleted());
-                                        if (!UpdatePrecentComplete(progressMonitor, 100 * nResolved++ / nUnresolved, ref progressStatus))
-                                            return null;
-                                        proteinMetadata = null;  // No search to be done
+                                        // Use Regexes to get some metadata, and a search term
+                                        var parsedProteinMetaData = FastaImporter.ParseProteinMetaData(proteinMetadata);
+                                        if ((parsedProteinMetaData == null) || Equals(parsedProteinMetaData.Merge(proteinMetadata), proteinMetadata.SetWebSearchCompleted()))
+                                        {
+                                            // That didn't parse well enough to make a search term, or didn't add any new info - just set it as searched so we don't keep trying
+                                            _processedNodes.Add(node.Id.GlobalIndex, proteinMetadata.SetWebSearchCompleted());
+                                            if (!UpdatePrecentComplete(progressMonitor, 100 * nResolved2 / nUnresolved, ref progressStatus2))
+                                                return;
+                                            proteinMetadata = null; // No search to be done
+                                        }
+                                        else
+                                        {
+                                            proteinMetadata = proteinMetadata.Merge(parsedProteinMetaData); // Fill in any gaps with parsed info
+                                        }
                                     }
-                                    else
+
+                                    if (proteinMetadata != null)
                                     {
-                                        proteinMetadata = proteinMetadata.Merge(parsedProteinMetaData);  // Fill in any gaps with parsed info
+                                        // We note the sequence length because it's useful in disambiguating search results
+                                        proteinsToSearch.Add(new ProteinSearchInfo(new DbProteinName(null, proteinMetadata), node.PeptideGroup.Sequence == null ? 0 : node.PeptideGroup.Sequence.Length));
+                                        docNodesWithUnresolvedProteinMetadata.Add(proteinsToSearch.Last(), node);
                                     }
                                 }
-                                if (proteinMetadata != null)
+
+                                if (node.ProteinMetadata is ProteinGroupMetadata proteinGroupMetadata)

```

Comment:
Not a fan of the heavy use of "is" to get to the list of metadata objects. Use a virtual instead and just have the single object return itself. You have derivation and virtuals.